test/Microsoft.Azure.SignalR.Common.Tests/RestClients/HealthApiFacts.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.SignalR.Tests;
+using Microsoft.Rest;
+using Xunit;
+
+namespace Microsoft.Azure.SignalR.Common.Tests.RestClients
+{
+    public class HealthApiFacts
+    {
+        [Fact]
+        public async void IsServiceHealthyReturnTrue()
+        {
+            using var signalRServiceRestClient = new TestRestClient(HttpStatusCode.OK);
+            var healthApi = new HealthApi(signalRServiceRestClient);
+
+            var operationResponse = await healthApi.GetHealthStatusWithHttpMessagesAsync();
+
+            Assert.Equal(HttpStatusCode.OK, operationResponse.Response.StatusCode);
+        }
+
+        [Theory]
+        [InlineData(HttpStatusCode.ServiceUnavailable)] //will retry many times
+        [InlineData(HttpStatusCode.GatewayTimeout)]  //will retry many times
+        [InlineData(HttpStatusCode.BadRequest)]  //won't retry
```

Comment:
how ut detects if it retries?

Reply:
No need to detect retrying. This test just ensure that exception is thrown given HTTP status code.